docs/access_and_permissions.md

```diff
@@ -0,0 +1,288 @@
+
+### Policies
+
+| Class                                                                                                       | Responsibility                                                                   |
+|-------------------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------------|
+| [Account::ApplicationPolicy](/app/policies/account/application_policy.rb)                                   | Determining whether granters can see and update their own access and permissions |
+| [SupportedPermissionPolicy](/app/policies/supported_permission_policy.rb)                                   | Determining which permissions can be updated by a given granter                  |
+| [UserPolicy](/app/policies/user_policy.rb)                                                                  | Determining whether a granter can update a grantee's access and permissions*     |
+| [UserApplicationPermissionPolicy](/app/policies/user_application_permission_policy.rb)                      | Determining whether a granter can update a grantee's access and permissions*     |
+| [UserPermissionManageableApplicationPolicy](/app/policies/user_permission_manageable_application_policy.rb) | Determing the apps to/for which a granter can manage access and permissions      |
+
+\* the responsibility of these two policies is hard to distinguish in this context, but as seen in the dependency trees, the `UserApplicationPermissionPolicy` depends on the `UserPolicy`, and in reality the latter is larger in scope
+
+### Others
+
+| Class                                                                               | Responsibility                                                                                 |
+|-------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------|
+| [ApplicationTableHelper](/app/helpers/application_table_helper.rb)                  | Generating links to view/manage access and permissions dependent on policy-based authorisation |
+| [SupportedPermissionParameterFilter](/lib/supported_permission_parameter_filter.rb) | Ensuring granters can only change permissions they're authorised to manage                     |
+| [UserUpdate](/app/services/user_update.rb)                                          | Updating a user's permissions                                                                  |
+| [UserUpdatePermissionBuilder](/app/models/user_update_permission_builder.rb)        | Generating a list of updated permissions                                                       |
+
+## To do
+
+- [ ] Cover how the signin permissions routes work, which are used in the `ApplicationTableHelper`
```

Comment:
Probably best to remove this TODO from here and make sure it's covered by a card on our Trello board.